homeassistant/components/filesize/sensor.py

```diff
@@ -12,69 +12,74 @@
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from homeassistant.helpers.reload import setup_reload_service
 from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
 
-from . import DOMAIN, PLATFORMS
+from .const import CONF_FILE_PATHS, DOMAIN
 
 _LOGGER = logging.getLogger(__name__)
 
-
-CONF_FILE_PATHS = "file_paths"
 ICON = "mdi:file"
 
 PLATFORM_SCHEMA = PARENT_PLATFORM_SCHEMA.extend(
     {vol.Required(CONF_FILE_PATHS): vol.All(cv.ensure_list, [cv.isfile])}
 )
 
 
-def setup_platform(
+async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
-    add_entities: AddEntitiesCallback,
+    async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the file size sensor."""
-
-    setup_reload_service(hass, DOMAIN, PLATFORMS)
-
-    sensors = []
-    paths = set()
+    _LOGGER.warning(
+        # Filesize config flow added in 2022.4 and should be removed in 2022.8
+        "Loading filesize via platform setup is deprecated; Please remove it from your configuration"
+    )
     for path in config[CONF_FILE_PATHS]:
-        try:
-            fullpath = str(pathlib.Path(path).absolute())
-        except OSError as error:
-            _LOGGER.error("Can not access file %s, error %s", path, error)
-            continue
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": SOURCE_IMPORT},
+                data={CONF_FILE_PATH: path},
+            )
+        )
 
-        if fullpath in paths:
-            continue
-        paths.add(fullpath)
 
-        if not hass.config.is_allowed_path(path):
-            _LOGGER.error("Filepath %s is not valid or allowed", path)
-            continue
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the platform from config entry."""
 
-        sensors.append(Filesize(fullpath))
+    path = entry.data[CONF_FILE_PATH]
+    try:
+        get_path = await hass.async_add_executor_job(pathlib.Path, path)
+        fullpath = str(get_path.absolute())
+    except OSError as error:
+        _LOGGER.error("Can not access file %s, error %s", path, error)
+        return
 
-    if sensors:
-        add_entities(sensors, True)
+    async_add_entities([FilesizeEntity(fullpath, entry.entry_id)], True)
 
 
-class Filesize(SensorEntity):
+class FilesizeEntity(SensorEntity):
     """Encapsulates file size information."""
 
     _attr_native_unit_of_measurement = DATA_MEGABYTES
     _attr_icon = ICON
 
-    def __init__(self, path: str) -> None:
+    def __init__(self, path: str, entry_id: str) -> None:
         """Initialize the data object."""
         self._path = path  # Need to check its a valid path
         self._attr_name = path.split("/")[-1]
+        self._attr_unique_id = entry_id
 
```

Comment:
Not for this PR, but it would be good to switch this to  Device (Service in this case) and separate `bytes` and `last_updated` extra_state_attributes into their own disabled by default sensors (see the tip at the end of this table https://developers.home-assistant.io/docs/core/entity/sensor?_highlight=sensor#properties)

Reply:
Makes sense. Can do that in follow-up PR